OpenRA.Mods.Common/Traits/Upgrades/UpgradeOnDamage.cs

```diff
@@ -35,18 +35,37 @@ public class UpgradeOnDamageInfo : ITraitInfo, Requires<UpgradeManagerInfo>
 		public object Create(ActorInitializer init) { return new UpgradeOnDamage(init.Self, this); }
 	}
 
-	public class UpgradeOnDamage : INotifyDamageStateChanged
+	public class UpgradeOnDamage : INotifyDamageStateChanged, INotifyCreated
 	{
 		readonly UpgradeOnDamageInfo info;
 		readonly UpgradeManager um;
 		bool granted;
+		Health health;
 
 		public UpgradeOnDamage(Actor self, UpgradeOnDamageInfo info)
 		{
 			this.info = info;
 			um = self.TraitOrDefault<UpgradeManager>();
 		}
 
+		void INotifyCreated.Created(Actor self)
+		{
+			health = self.TraitOrDefault<Health>();
```

Comment:
Can we put this in the constructor and add `Requires<HealthInfo>`?  It makes zero sense for an actor without Health to implement this trait.